connect/runtime/src/test/java/org/apache/kafka/connect/runtime/ExactlyOnceWorkerSourceTaskTest.java

```diff
@@ -483,6 +494,37 @@ public void testFailureInPoll() throws Exception {
         assertPollMetrics(0);
     }
 
+    @Test
+    public void testRetriableExceptionInPoll() throws Exception {
+
+        final ErrorHandlingMetrics errorHandlingMetrics = mock(ErrorHandlingMetrics.class);
+        final RetryWithToleranceOperator retryWithToleranceOperator = new RetryWithToleranceOperator(30, 15, ALL, SYSTEM, errorHandlingMetrics);
```

Comment:
Shouldn't we use `MockTime` here so that we can advance the time programatically on each call to poll and ensure the retries occur as per the deadline?

Reply:
I am using the existing mechanisms to run the test using CountDownLatch. I could get the enough number of invocations but will think about it.

Reply:
That's probably due to https://github.com/apache/kafka/pull/13726#discussion_r1214152577 (i.e. poll being called repeatedly in the task execution loop)?

Reply:
yeah. I guess that's expected as explained already [here](https://github.com/apache/kafka/pull/13726#discussion_r1218190316)?